samza-core/src/main/java/org/apache/samza/execution/JobNode.java

```diff
@@ -92,14 +93,21 @@ void addOutEdge(StreamEdge out) {
     return outEdges;
   }
 
-  public JobConfig generateConfig() {
+  /**
+   * Generate the configs for a job
+   * @param executionPlanJson JSON representation of the execution plan
+   * @return config of the job
+   */
+  public JobConfig generateConfig(String executionPlanJson) {
```

Comment:
I don't understand this method signature. Why should generateJobConfig take executionPlanJson? Also, why is it being added to the config?